### PR TITLE
Minor comment fixes

### PR DESCRIPTION
**Pull request by danra**
#### What's in this pull request?

---

Before merging this pull request to apple/swift repository:
- [ ] Test pull request on Swift continuous integration.

<details>
  <summary>

Triggering Swift CI</summary>

The swift-ci is triggered by writing a comment on this PR addressed to the GitHub user @swift-ci. Different tests will run depending on the specific comment that you use. The currently available comments are:

**Smoke Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please smoke test |
| OS X platform | @swift-ci Please smoke test OS X platform |
| Linux platform | @swift-ci Please smoke test Linux platform |

 **Validation Testing**

| Platform | Comment |
| --- | --- |
| All supported platforms | @swift-ci Please test |
| OS X platform | @swift-ci Please test OS X platform |
| Linux platform | @swift-ci Please test Linux platform |

Note: Only members of the Apple organization can trigger swift-ci.
</details>

### CODE REVIEW

File: include/swift/Basic/Malloc.h [hunk 1]
```diff
@@ -44,4 +44,4 @@ inline void AlignedFree(void *p) {
   
 }
 
-#endif
+#endif /* __SWIFT_MALLOC_H__ */
```
Comment (gribozavr) [inline include/swift/Basic/Malloc.h]: While you are here, could you change the include guard to be `SWIFT_BASIC_MALLOC_H`?

Comment (danra) [inline include/swift/Basic/Malloc.h]: Done

**Comment by gribozavr** [comment]
@swift-ci Please test

File: include/swift/Basic/Malloc.h (hunk 1 above)
Comment (gottesmm) [inline include/swift/Basic/Malloc.h]: Also it should be using //, not /*.

Please see:

http://llvm.org/docs/CodingStandards.html#comment-formatting

File: include/swift/Basic/Malloc.h [hunk 2]
```diff
@@ -44,4 +44,4 @@ inline void AlignedFree(void *p) {
   
 }
 
-#endif /* __SWIFT_MALLOC_H__ */
+#endif /* SWIFT_BASIC_MALLOC_H */
```
Comment (gottesmm) [inline include/swift/Basic/Malloc.h]: Can you change this to use a C++ style commit ala the LLVM style guide:

http://llvm.org/docs/CodingStandards.html

[btw, thanks for making these changes, this is an iterative process of you learning the style that is used in our project = )]

File: include/swift/Basic/Malloc.h (hunk 1 above)
Comment (danra) [inline include/swift/Basic/Malloc.h]: ...unless, from the attached link: "When writing a header file that may be #included by a C source file."

Specifically, I thought it's a better convention to use /* */ comment in the last line of the file, for the unfortunate case when a newline is missing at the end of the file, which means a // comment could comment content following the #include of the header.

Comment (danra) [inline include/swift/Basic/Malloc.h]: Never mind, looks like this is a non-issue in C++11. Will change.

Comment (gottesmm) [inline include/swift/Basic/Malloc.h]: No worries. This is a learning process. How would you know otherwise if you are starting to contribute?

**Comment by gottesmm** [comment]
Can you rebase/squash this please.

**Comment by danra** [comment]
Squashed

**Comment by gottesmm** [comment]
This LGTM.

**Pull request merged**